### DISCUSSION TITLE
apm listing different version of npm than my system

### DISCUSSION THREAD

**Discussion by bassamanator** [atom/atom#23032] [Q&A]
Hi there,

Atom terminal programs are using an older version of npm than my system.

In atom terminal (termination):
```
$ npm --version
6.14.8
```

In my system terminal:
```
$ npm --version
7.24.0
$ apm --version
apm  2.6.4
npm  6.14.9
node 16.9.1 x64
atom 1.58.0
python 3.9.7
git 2.33.0
```

Is there someway to change the version of npm used in atom/apm?

Thanks!

> Atom    : 1.58.0
Electron: 9.4.4
Chrome  : 83.0.4103.122
Node    : 12.14.1
Operating System: Manjaro Linux
KDE Plasma Version: 5.22.5
KDE Frameworks Version: 5.85.0
Qt Version: 5.15.2
Kernel Version: 5.11.4-1-rt11-MANJARO (64-bit)
Graphics Platform: X11
Processors: 16 × AMD Ryzen 7 2700X Eight-Core Processor
Memory: 31.4 GiB of RAM
Graphics Processor: Radeon RX 580 Series

**Comment by icecream17** [comment]
On discuss.atom.io I asked if there was a way to update the apm version, and someone answered that apm is bundled with Atom so there's isn't really a way to update it manually.

But you can update the (presumably cached) version by reinstalling atom

**Reply by bassamanator** [reply]
Thanks but I'm interested in changing the version of **npm** that atom/apm use, the version of **apm** is of no import to me.

**Reply by icecream17** [reply]
It updates all the versions

**Reply by bassamanator** [reply]
So, do you think if I just uninstall atom, and then just reinstall it, it will update the version of npm that atom uses?

**Reply by bassamanator** [reply]
I uninstalled atom. *Did not delete .atom/* because that would be like commiting suicide right now. Clean my package manager cache, meaning that the atom package would need to be downloaded before install. Installed atom, and unfortunately, no version numbers have change; still using npm 6.14.8 in the atom terminal whereas my system itself has npm 7.24.0 installed.

:/
